### PR TITLE
FossGraph. Improved Foss Graph view

### LINKED ISSUE #2098: FossGraph. Improve Foss Graph view

### PR DESCRIPTION
**Pull request by Cheshiriks**
Снимок

Closes #2098

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+package com.saveourtool.save.frontend.components.basic
+
+import csstype.ClassName
+import react.ChildrenBuilder
+import react.dom.html.ReactHTML
+
+private const val SAVE_FOSS_GRAPH_INTRO_MD = """
+    |## save-foss-graph
+    |It is a list of publicly disclosed information security vulnerabilities and exposures.
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt:13]:
```suggestion
    |This is a list of publicly disclosed information security vulnerabilities and exposures.
```

Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt:13]:
```suggestion
    |Current page provides the list of publicly disclosed information security vulnerabilities and exposures.
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        ReactHTML.div {
+            className = ClassName("card-body d-flex align-items-start")
+            ReactHTML.div {
+                ReactHTML.strong {
+                    className = ClassName("d-inline-block mb-2 text-info")
+                    +"Introducing"
+                }
+                markdown(SAVE_FOSS_GRAPH_INTRO_MD.trimMargin())
+            }
+            ReactHTML.div {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt:45]:
```suggestion
            div {
```
and so on

copy-paste kills

**Review by sanyavertolet — COMMENTED** [review]
Talking about the screenshot, I think these two cards should have the same height

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package com.saveourtool.save.frontend.components.basic
+
+import csstype.ClassName
+import react.ChildrenBuilder
+import react.dom.html.ReactHTML
+
+private const val SAVE_FOSS_GRAPH_INTRO_MD = """
+    |## save-foss-graph
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt:12]: may be just 

```suggestion
    |## FOSS graph
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    |It is a list of publicly disclosed information security vulnerabilities and exposures.
+"""
+
+private const val SAVE_FOSS_GRAPH_ADD_NEW_MD = """
+    |### New vulnerability
+    |You can suggest your own new vulnerability if you didn't find the required one in our list.
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt:18]:
```suggestion
    |You can suggest your own new vulnerability, if you didn't find the required one in our list.
```

**Comment by Cheshiriks** [comment]
> Talking about the screenshot, I think these two cards should have the same height

For the right card, the value depends on the number of rows in the table. It seemed to me that the option with just a lot of white background is even less beautiful.

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    |After our approval, it will appear in the database.
+"""
+
+private const val SAVE_FOSS_GRAPH_HOW_TO_MD = """
+    |### How to add vulnerability in project?
+    |1. Create saveourtool [organization](#/create-organization) and [project](#/create-project);
+    |2. Go to your project's security tab;
+    |3. Create new problem and add vulnerability number;
+    |
+    |In case of any error feel free to [contact us](https://github.com/saveourtool/save-cloud/issues/new).
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt:28]: Also please add contancts for creating the vulnerability here like `[saveourtool@gmail.com](mailto:saveourtool@gmail.com)`

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphCollectionView.kt [hunk 6]
```diff
@@ -97,46 +100,83 @@ val fossGraphCollectionView: FC<FossGraphCollectionViewProps> = FC { props ->
...
-                post(
-                    url = "$apiUrl/vulnerabilities/by-filters",
-                    headers = jsonHeaders,
-                    body = Json.encodeToString(vulnerabilityFilters),
-                    loadingHandler = ::noopLoadingHandler,
-                    responseHandler = ::noopResponseHandler,
-                ).unsafeMap {
-                    it.decodeFromJsonString()
+                        div {
+                            @Suppress("TOO_MANY_LINES_IN_LAMBDA")
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphCollectionView.kt:132]: can we configure this rule somehow to not to trigger on frontend?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt (hunk 4 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FossGraphIntroductionComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphCollectionView.kt (hunk 6 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphCollectionView.kt:132]: We can only adjust the maximum size of the lamda itself.
We can create an issue on Diktat.

**Comment by sanyavertolet** [comment]
> > Talking about the screenshot, I think these two cards should have the same height
> 
> For the right card, the value depends on the number of rows in the table. It seemed to me that the option with just a lot of white background is even less beautiful.

In this case, the right card shouldn't be so big that it would cause a lot of white background in the left card

**Pull request merged**